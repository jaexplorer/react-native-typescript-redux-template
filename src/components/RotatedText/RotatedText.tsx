/* eslint-disable react/no-array-index-key */
import React, { FC } from 'react';
import { View } from 'react-native';
import { useTheme } from '../../lib/Theme';
import Text from '../Text';
import stylesCreator from './RotatedTextStyles';

const RotatedText: FC<RotatedTextProps> = ({ text }) => {
  const textArray = text.split('');
  const [styles] = useTheme(stylesCreator);

  return (
    <View style={styles.textArray}>
      {textArray.map((t, i) => (
        <View key={i} style={styles.letterWrapper}>
          <Text bold style={styles.text}>
            {t}
          </Text>
        </View>
      ))}
    </View>
  );
};

export interface RotatedTextProps {
  text: string;
}

export default RotatedText;
