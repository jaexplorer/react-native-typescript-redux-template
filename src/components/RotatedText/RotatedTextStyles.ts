import { StyleSheet } from 'react-native';
import { StylesCreator } from '../../lib/Theme';

const stylesCreator: StylesCreator = (theme, scale, moderateScale) =>
  StyleSheet.create({
    textArray: {
      width: scale(6),
    },
    letterWrapper: {
      width: scale(6),
      height: scale(4.5), // Padding between each letter
      alignItems: 'center',
      justifyContent: 'space-around',
    },
    text: {
      transform: [
        {
          rotate: '90deg',
        },
      ],
      textTransform: 'uppercase',
      fontSize: moderateScale(9),
    },
  });

export default stylesCreator;
