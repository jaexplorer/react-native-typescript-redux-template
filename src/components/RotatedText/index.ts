import RotatedText from './RotatedText';

export default RotatedText;
