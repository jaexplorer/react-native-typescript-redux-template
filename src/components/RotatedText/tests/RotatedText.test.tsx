import React from 'react';
import { render } from '@testing-library/react-native';
import RotatedText from '../RotatedText';

test('Does component render test', () => {
  const { baseElement } = render(<RotatedText text="Test" />);
  expect(baseElement).toMatchSnapshot();
});
