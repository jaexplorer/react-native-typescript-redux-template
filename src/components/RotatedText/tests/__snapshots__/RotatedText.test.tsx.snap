// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does component render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "width": 12.857142857142858,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 9.642857142857142,
            "justifyContent": "space-around",
            "width": 12.857142857142858,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1}
          style={
            Array [
              Object {
                "bottom": -5,
                "color": "#999999",
                "fontSize": 22,
                "marginBottom": 4.285714285714286,
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "fontSize": 14.142857142857142,
                "textTransform": "uppercase",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        >
          T
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 9.642857142857142,
            "justifyContent": "space-around",
            "width": 12.857142857142858,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1}
          style={
            Array [
              Object {
                "bottom": -5,
                "color": "#999999",
                "fontSize": 22,
                "marginBottom": 4.285714285714286,
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "fontSize": 14.142857142857142,
                "textTransform": "uppercase",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        >
          e
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 9.642857142857142,
            "justifyContent": "space-around",
            "width": 12.857142857142858,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1}
          style={
            Array [
              Object {
                "bottom": -5,
                "color": "#999999",
                "fontSize": 22,
                "marginBottom": 4.285714285714286,
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "fontSize": 14.142857142857142,
                "textTransform": "uppercase",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        >
          s
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 9.642857142857142,
            "justifyContent": "space-around",
            "width": 12.857142857142858,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={1}
          style={
            Array [
              Object {
                "bottom": -5,
                "color": "#999999",
                "fontSize": 22,
                "marginBottom": 4.285714285714286,
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "fontSize": 14.142857142857142,
                "textTransform": "uppercase",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              },
            ]
          }
        >
          t
        </Text>
      </View>
    </View>
  </View>
</View>
`;
